### ISSUE TITLE
Why not use jq?

### ISSUE THREAD

**Issue by justsaumit** [swindlesmccoop/weatheralertd#1]
Hey there! Lovely script, certainly very based.

However, as the title suggests, have you considered using jq for parsing JSON data? 
jq is a powerful JSON parser that simplifies the process of extracting specific fields from JSON data in a v comfy manner.

In case the original output of the API changes, you would have to manually grep and pipe which field to extract using awk, etc. With jq, it's all simplified and shortened as follows:

``` bash
REGION="$(curl -s https://ipapi.co/json/ | jq -r .region_code)"
URL="https://api.weather.gov/alerts/active/area/$REGION"
SLEEPTIME=60

_getalert() {
	curl -s $URL |  jq -r '.features[0].properties.parameters.NWSheadline[0]'
```
Rest of the code can stay pretty much the same!

I would have made a PR, but I'd like to know your thoughts. Have you considered it? Is it slow, bloated, or another dependency? Or maybe you agree that it's actually a better replacement?
Let me know and I hope this helps!

**Comment by swindlesmccoop** [comment]
I actually was using `jq`, which changed on commit https://github.com/swindlesmccoop/weatheralertd/commit/271e92fff33366780449e377fa2f1906adc8c307. I changed it because I wanted to reduce dependencies. If you want to implement it and run `time` and can prove that `jq` is faster than what I'm doing then I'll merge a PR using it.

**Comment by justsaumit** [comment]
Oh my bad should've checked past commits. 
I actually did some trials using `time` and for loop and found inconsistent results, so I removed `curl` and stored the output as a JSON file.  Through which I clearly found who's the the winner

![image](https://github.com/swindlesmccoop/weatheralertd/assets/79687674/5568f3bf-bfff-48d4-b883-71acb74debd2)


This could be because `jq` operates on the JSON data as a whole, then parsing through the layers.. meanwhile `grep` uses straightforward pattern matching to extract specific lines.
I used `cat` instead of directly piping to have a even playing field.
So in this particular case, it seems that `grep` performs faster than `jq`, so it is more favorable to prioritize performance over simplicity. 
I will close the issue. Thank you!
